libphal/phal_dump.C

```diff
@@ -101,16 +101,19 @@ struct DumpPPERegValue {
 /**
  * @brief Return the proc target corresponding to failing unit
  * @param[in] failingUnit Position of the proc containing the failed SBE
+ * @param[in] forOdySsey Proc target type i.e.; Odyssey or normal memory chip
  *
  * @return Pointer to the pdbg target for proc
  *
  * Exceptions: PDBG_TARGET_NOT_OPERATIONAL if target is not available
  */
-struct pdbg_target* getProcFromFailingId(const uint32_t failingUnit)
+struct pdbg_target* getProcFromFailingId(const uint32_t failingUnit, const bool forOdySsey)
 {
-	struct pdbg_target* proc = NULL;
-	struct pdbg_target* target;
-	pdbg_for_each_class_target("proc", target)
+	struct pdbg_target* proc = nullptr;
+	struct pdbg_target* target = nullptr;
+	const char* procTypeString = (forOdySsey ? "membuf" : "proc");
```

Comment:
Variable name procTypeString is not correct, membuf is not a proc

Reply:
Corrected